platforms/jvm/plugins-java/src/integTest/groovy/org/gradle/java/JavaCrossCompilationIntegrationTest.groovy

```diff
@@ -16,148 +16,62 @@
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.integtests.fixtures.jvm.JavaToolchainFixture
 import org.gradle.integtests.fixtures.jvm.TestJavaClassUtil
 import org.gradle.internal.FileUtils
-import org.gradle.internal.jvm.Jvm
-import org.gradle.test.fixtures.Flaky
+import org.gradle.internal.serialize.JavaClassUtil
 import org.gradle.test.fixtures.archive.JarTestFixture
-import org.gradle.test.fixtures.file.DoesNotSupportNonAsciiPaths
-import org.gradle.util.GradleVersion
-import org.junit.Assume
 
-import static org.gradle.internal.serialize.JavaClassUtil.getClassMajorVersion
-
-@DoesNotSupportNonAsciiPaths(reason = "Java 6 seems to have issues with non-ascii paths")
-@Flaky(because = "https://github.com/gradle/gradle-private/issues/3901")
-class JavaCrossCompilationIntegrationTest extends AbstractIntegrationSpec {
-
-    static List<String> javaVersionsWithoutTestingSupport() {
-        return ["1.6", "1.7"]
-    }
-
-    static List<String> fullySupportedJavaVersions() {
-        // Keep this to LTS versions, and latest
-        return ["1.8", "11", "17", "21", "24"]
-    }
-
-    static List<String> allJavaVersions() {
-        return fullySupportedJavaVersions() + javaVersionsWithoutTestingSupport()
-    }
-
-    static JavaVersion toJavaVersion(String version) {
-        return JavaVersion.toVersion(version)
-    }
+/**
+ * Tests java compilation across JDK versions.
+ */
+class JavaCrossCompilationIntegrationTest extends AbstractIntegrationSpec implements JavaToolchainFixture {
 
-    Jvm withJavaProjectUsingToolchainsForJavaVersion(String version) {
-        def javaVersion = toJavaVersion(version)
-        def target = AvailableJavaHomes.getJdk(javaVersion)
-        Assume.assumeNotNull(target)
-        withJavaProjectUsingToolchainsForJavaVersion(target)
-        target
-    }
+    def "can compile java source using java #jdk.javaVersionMajor"() {
+        given:
+        settingsFile << """
+            rootProject.name = 'foo'
+        """
 
-    def withJavaProjectUsingToolchainsForJavaVersion(Jvm jvm) {
         buildFile << """
             plugins {
-                id("java")
+                id("java-library")
             }
 
             ${mavenCentralRepository()}
-
-            java {
-                toolchain {
-                    languageVersion = JavaLanguageVersion.of(${jvm.javaVersion.majorVersion})
-                }
-            }
-
-            tasks.withType(Javadoc) {
-                options.noTimestamp = false
-            }
+            ${javaPluginToolchainVersion(jdk)}
         """
 
         file("src/main/java/Thing.java") << """
-            /** Some thing. */
             public class Thing { }
         """
 
-        executer.withArgument("-Porg.gradle.java.installations.paths=" + jvm.javaHome.absolutePath)
-    }
-
-    def "can compile source and run JUnit tests using target Java version"() {
```

Comment:
JUnit and TestNG test coverage has been moved to `AbstractTestJavaVersionIntegrationTest`